ayeletK/assignment1/Q1_solution.cpp

```diff
@@ -0,0 +1,114 @@
+#include <string>
+#include <vector>
+#include <cctype>
+#include <iostream>
+#include <algorithm>
+
+using namespace std;
+
+/**
+    Determine if a character is a punctuation mark
+	  from the next list: ‘ ‘/’,’/’.’/’!’/’?’
+    @param c - the character to be checked
+    @return true if the character is a punctuation mark,
+	    false otherwise.
+    
+    example:
+    c = "!"
+    ------------------
+    true
+*/
+bool is_punctuation_mark(char c){
+    if (c == ' ' || c == ',' || c == '.' ||
+        c == '!' || c == '?'){
+            return true;
+        }
+    return false;
+}
+
+/**
+    Split a sentence by white spaces and punctuation marks
+    [',','.','!','?'] into a vector and sort each word 
+    ("aba" --> "aab")
+
+    @param str - the sentence to be splitted
+    @param case_sensitive - a flag determins if we are 
+      working in a case sensitive mode or not.
+      if not 'case_sensitive' then all letters will be 
+      lowered to lower case.
+    @return a vector of strings, each string in the result
+      vector represent a sorted word from the original
+      sentence.
+    
+    example:
+    str = "Like a G6"
+    case_sensitive = false
+    ------------------
+    words: <"eikl", "a", "6g">
+*/
+vector<string> split_to_words(string str, bool case_sensitive){
+    vector<string> words;
+    string curr_str = "";
```

Comment:
`string curr_str;` is enough, a string is defaultly initialized to an empty string.